packages/shared/lib/services/connection.service.ts

```diff
@@ -658,16 +673,31 @@ class ConnectionService {
                 await logCtx.error('Failed to refresh credentials', error);
                 await logCtx.failed();
 
-                // TODO now insert into notifications to recall this error and link to it
+                if (activityLogId) {
+                    await errorNotificationService.auth.create({
+                        type: 'auth',
+                        action: 'token_refresh',
+                        connection_id: connection.id,
+                        activity_log_id: activityLogId,
+                        log_id: logCtx.id,
+                        active: true
+                    });
+                }
```

Comment:
It looks like we are now duplicating some activity logs so they can be used in a different way. 
I understand we cannot used activity logs directly because of the need for a longer retention period but instead of inventing a new model/logic would it make sense to implement a mechanism to persist some of `logCtx` data to the db?
So instead of duplicating the object being saved we call something like `persistLongTerm(logCtx)`. This single function would then transform the data to whatever format we want and save it to the db. That could  greatly improve the reusability and prevent the need to create new function every time we want to add another error type.

Feel free to ping me if you want to discuss about this topic.

Reply:
We discussed this with Thomas, I agree we need some kind of way to limit the number of things doing similar stuff (same problem with error reporting in general). Ideally I think we would have an event stream where we subscribe to what's interesting to each system, in the meantime I'm fine with the current state. 

Note: I didn't realized during my review but the proposed system (storing the activityLogId) is not playing well with the retention of 15days (old and new logs system), so it might indeed need some change unless we are fine with it.

Reply:
So we're not duplicating logs but rather linking to them. So a reference to the log is stored but the error isn't duplicated anywhere. The 15 day retention shouldn't be an issue for auth refreshes since stale tokens refresh every day so there should always be a fresh log.

If a sync doesn't run for 15 days it is dormant anyway so don't see an issue there really. Also, if a log is deleted with cascading the entry in the active log table is also deleted.